lab_2/main.py

```diff
@@ -81,7 +183,36 @@ def calculate_text_plagiarism_score(original_text_tokens: tuple, suspicious_text
     :param plagiarism_threshold: a threshold
     :return: a score from 0 to 1, where 0 means no plagiarism, 1 – the texts are the same
     """
-    pass
+    orig_txt_not_tuple = not isinstance(original_text_tokens, tuple)
+    sus_text_not_tuple = not isinstance(suspicious_text_tokens, tuple)
+    plagiarism_not_float = not isinstance(plagiarism_threshold, float)
+
+    for text_tokens in (original_text_tokens, suspicious_text_tokens):
+        if isinstance(text_tokens, tuple):
+            for _, token in enumerate(text_tokens):
+                if isinstance(token, tuple):
+                    if not all(isinstance(word, str) for word in token):
+                        return -1.0
+                else:
+                    return -1.0
+
+    if orig_txt_not_tuple or sus_text_not_tuple \
+            or plagiarism_not_float or 0 > plagiarism_threshold > 1:
+        return -1.0
+
+    original_list = list(original_text_tokens)
+    if len(original_text_tokens) < len(suspicious_text_tokens):
+        original_list.append(()*(len(suspicious_text_tokens) - len(original_text_tokens)))
+    elif len(suspicious_text_tokens) < len(original_text_tokens):
```

Comment:
do you need this?

Reply:
same here, i do this so the code can handle different sized inputs